### ISSUE TITLE
Support multiple accounts

### ISSUE THREAD

**Issue by stefan-niedermann** [deltachat/deltachat-android#18]
Would be nice if one could use multiple email accounts.

**Comment by r10s** [comment]
This is planned, in some way. Esp. to use completely independent private and business account.

**Comment by MichaIng** [comment]
Moreover multiple identities for the same account would be awesome. E.g. I use separate addresses on the same account for different use cases, e.g. private and online accounts with spam risk.

**Comment by meitar** [comment]
I second what @MichaIng says:

> Moreover multiple identities for the same account would be awesome.

This is also a common thing among GMail users who tend to use the `+` separator and has been [an official "tip" promoted by Google for a long time](https://gmail.googleblog.com/2008/03/2-hidden-ways-to-get-more-from-your.html).

While I know that's GMail-specific, it would be neat if Delta Chat recognized this technique or optionally provided a way for me to signal to the app that I want to treat `+`-delimited addresses as separate chat identities.

Another way to do this that is less GMail-specific may to append a header (`X-DELTA-CHAT-IDENTITY` maybe?) to the headers, although this does leak some metadata as OpenPGP cannot encrypt headers, AFAIK.

**Comment by fauno** [comment]
my 2 cents: it's not gmail specific, it's at least supported by postfix, where it's called recipient delimiter.

it'd be good to support discardable email accounts, though i don't know if any of these services provide imap access

**Comment by haeckser** [comment]
A simple solution to allow usage of multiple accounts one by one (not in parallel) can be the usage and switching between separate databases (one DB per Account as it is designed today)
If the standard database is extended with the names of alternative DBs and the related accounts, the UI (Android) can close the DB and open another one. There is no need to change the server, just add  another UI item to show the alternative accounts and to make the switch. IF there is a name sceme for alternative DBs switching between teh accounts might be possible between all different accounts. Otherwise there is only the way to switch back to the standard account and from there to another.
Everything else is compatible with the current versions of UI and core.

**Comment by r10s** [comment]
Duplicate of https://github.com/deltachat/deltachat-core/issues/25

**Comment by r10s** [comment]
there were some recent discussions in the core, https://github.com/deltachat/deltachat-core/issues/25#issuecomment-381123989 however, as the core already supports multiple accounts, i closed the core-issue and re-opened this one.

**Comment by hpk42** [comment]
IMO the main issue about multi-account is the UI design and implementation of it.  If someone can provide UI mocks on how it should work (setup, account selection etc.) please post .

**Comment by mbeko** [comment]
I'll provide my personal ideas based on my needs, maybe that can provide some orientation. In general, the XMPP client Conversations is a good example of a multi-account UI.

I imagine the account management view as in Conversations: You can add accounts to the list and activate or deactivate them separately via switch on each entry.

The chat list would then display the chats for the selected accounts in a single list because for receiving I don't care much about to which account a message has been sent. So the same goes for contact requests.

For replies, the proper account would be automatically selected and for new chats I would select it myself.

Contacts would be independent of the account, I could always see all of them.

**Comment by thomaspeterthiede** [comment]
i am also for multiple accounts, where every account can be switch offline seperately like in conversations.

**Comment by r10s** [comment]
i'm closing this for now here; https://support.delta.chat is the better place for feature requests and discussions.

**Comment by adbenitez** [comment]
any plans of implementing this now that 0.100 is out??

**Comment by r10s** [comment]
nothing concrete, but sooner or later we will probably go for it.

the desktop app already supports multiple accounts (and deltachat-core as well).

**Comment by hpk42** [comment]
Likely it's not the next thing to do but yes, 
it's now more feasible to think about it ;)

**Comment by esmoak** [comment]
My Gmail account has a dot in it. Like blah.blah@gmail but Gmail treats that dot like a + to an extent. So some people email blahblah@gmail since I got lazy mentioning the dot. 

It sorta makes Delta Chat unusable for me as a main way of communicating if it can't let me receive messages for both ways the address is written. 

Love the program and I will keep it installed to hopefully still implement it in my daily use somehow.

**Comment by zgzgng** [comment]
@MichaIng 
> multiple identities for the same account would be awesome. E.g. I use separate addresses on the same account

I do not understand your use of the word 'account'.

**Comment by MichaIng** [comment]
@hpk42
I have multiple mail providers as well for different kind of contacts but usually on each of them multiple alias mail addresses for organisational reasons. To be true I haven't used delta chat for a while because of this, so not sure about meanwhile added features :wink:.

**Comment by hpk42** [comment]
On Wed, Feb 05, 2020 at 01:31 -0800, MichaIng wrote:
> @hpk42
> I have multiple mail providers as well for different kind of contacts but usually on each of them multiple alias mail addresses for organisational reasons. To be true I haven't used delta chat for a while because of this, so not sure about meanwhile added features :wink:.

From UX feedback gathered so far through 10-12 one-hour
interviews on this, most people use *different* apps for separating concerns. 

Even if we made aliases or some such configurable how would that show on
the UX side?  There are a lot of choices that people would probably want
to do, and that's a severe UX challenge -- every choice and decision
(or configuration) presented to unsuspecting users is challenging
and requires maintenance (FAQ entries, bug fixes, better onboarding ...)

**Comment by MichaIng** [comment]
@hpk42 
Yes, basically it would mean that every chat needs to be attached to a certain account and/or alias. A default could be implemented when opening new chats, but it needs to be possible to choose it of course. When receiving a mail/new message, the account/alias could be derived from the mail address, but can be attached only if this alias has been created already, else to be base account... yeah definitly a bunch of additional GUI work which would need to be implemented wisely to keep it clean not overloaded... I understand that this is out of scope as long as there not a large number of requesters or someone who contributes this feature to the code.

**Comment by hpk42** [comment]
On Wed, Feb 05, 2020 at 08:55 -0800, MichaIng wrote:
> @hpk42 
> Yes, basically it would mean that every chat needs to be attached to a certain account and/or alias. A default could be implemented when opening new chats, but it needs to be possible to choose it of course. When receiving a mail/new message, the account/alias could be derived from the mail address, but can be attached only if this alias has been created already, else to be base account... yeah definitly a bunch of additional GUI work which would need to be implemented wisely to keep it clean not overloaded... I understand that this is out of scope as long as there not a large number of requesters or someone who contributes this feature to the code.

heh, UX/implementation needed on three different code bases ... 
a more feasible step would be bringing what desktop does to mobile UX: 

- "switch account" from Chat-list would go to an account-list
- select account (or create a new one) switches to Chat-list 

this way there would be no new UX within current DC itself. 
And it's probably less confusing than mixed-account chat-list views.

**Comment by MichaIng** [comment]
@hpk42 
Sounds reasonable. However me personally would prefer a mixed-account chat-list indeed. Having to switch the account first, requires at least two additional touches every time, not very handy to quickly see/switch/join/create a chat. However I see the argument with cleaner UI...

**Comment by mbeko** [comment]
As there was not much of a reaction to my comment and the UI argument came up again, I'd like to [repeat it](https://github.com/deltachat/deltachat-android/issues/18#issuecomment-398535560):
> In general, the XMPP client Conversations is a good example of a multi-account UI.

I currently don't have the time, but if I finally get to help implement this, I'd basically copy that UI.

> From UX feedback gathered so far through 10-12 one-hour
interviews on this, most people use *different* apps for separating concerns.

Personally, I'd really like to have the possiblity of using multiple accounts because I don't separate by "chat vs. e-mail address" but by "public vs. private address". As mentioned in the second post here.

And because one of the advertised ideas behind Delta Chat is that you can use existing e-mail accounts, that's what I started doing. Now it would be quite painful to tell my contacts that they should please use a new address which I would have to create for chatting.

**Comment by thomaspeterthiede** [comment]
Hello mbeko,

that is my opinion too.
it is important for me tu use it with different account, not only alias.
conversations is the best example.
one account/mailbox for privat and one for busssines or other.
i am wondering that must be discuss, this feature hast to be a normal option.

i am speaking with other people who find delta chat interesting, but the missing option of multiple account is an exclusion for there.

greetings

**Comment by the-Arioch** [comment]
Since traditional mail programs are all by dead (still using Opera 12.x for it) i was thinking to try Delta Chat.

But it seems at max i will be able to switch only one of my mailboxes to Delta...

**Comment by hpk42** [comment]
On Sun, May 17, 2020 at 23:01 -0700, the-Arioch wrote:
> Since traditional mail programs are all by dead (still using Opera 12.x for it) i was thinking to try Delta Chat.
> 
> But it seems at max i will be able to switch only one of my mailboxes to Delta...

that's a good start ;) 

Is there anything in particular that blocks you? 

Or is it just that Delta Chat is not a full substitute for a regular e-mail app? 
It's indeed not our priority to make it one before we have a decent and 
reliable chat-mail app on all platforms. Some people are using as a substitute
but YMMV.

**Comment by mbeko** [comment]
I don't think this feature has much to do with Delta Chat being a mail program or not. Chat programs like Conversations also support using multiple accounts at the same time.

I want to use Delta Chat for chat only, I have several existing e-mail addresses and telling my contacts already using those to please use a new address would somehow collide with the idea of Delta Chat to pick up on the existing infrastructure.

**Comment by haeckser** [comment]
The current implementation of Delta Desktop works fine - so the backend is still able to support it on phones too. It is just the UI that must be enhanced (I expect in a a very easy way - one menu item, one selection dialog). If someone will implement a mixed account chatlist - fine - maybe it is another choice option in the account dialog and we have both behaviours without too much new implementations

**Comment by adbenitez** [comment]
deltachat-android does support multiple accounts like deltachat-desktop, what isn't supported yet is accounts working concurrently (only one account is active at the time)

about your situation, you could also set your other accounts to forward chats to your desired delta chat account, anyway it is better to use only one account instead of several accounts, that would mean more battery and network traffic usage, so letting your friends to know you now use a new email, isn't that hard and will have benefits

**Comment by mbeko** [comment]
I know all of that regarding the multiple account support. I just wanted to correct the perspective that this feature has anything to do with mail program features, and to explain why concurrent accounts are useful and how this is already solved well in other chat apps.

Thanks for the advice, but it somehow misses the point I made in my comment :wink:
